libnmstate/nm/connection.py

```diff
@@ -134,7 +134,9 @@ def set_master_setting(con_setting, master, slave_type):
 
 def get_device_connection(nm_device):
     act_connection = get_device_active_connection(nm_device)
-    if act_connection:
+    if (act_connection and
+            act_connection.props.state ==
+            nmclient.NM.ActiveConnectionState.ACTIVATED):
```

Comment:
https://github.com/nmstate/nmstate/pull/116#issuecomment-433741650
> > * Maybe add a test to check that profiles that are going to be activated are not reported
> 
> This is probably related to #120 , but I do not know how to check it in a stable manner: I can start a profile, but nothing assures it will still be there when we check it. Perhaps a dynamic IPv4 profile will do the job.
> 
> > * On first thought it sounds better to me to wait for the profile to be fully activated and report it then. Otherwise it seems to be a bad race condition.IMHO it would be better to track this problem somewhere.
> 
> Some connections we should wait for and some we should not (like the auto ones which nmstate did not asked for). The ones we do control, we need to wait for them using a callback: Opened https://nmstate.atlassian.net/browse/NMSTATE-67 to track it.

What does it mean not to wait for the automatically generated profiles? Wouldn't those be the profiles that we would want to change with nmstate? @cathay4t what is your opinion here?